From 9348fd0cb8990c6ea3e0efb13636171d2afe5bac Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 06 Jan 2022 07:33:38 +0200
Subject: [PATCH] [SW-66640] gaudi2: update reset cause before issuing reset

Currently in handling of hard reset in preboot, the reset cause
is updated after issuing reset which doesn't take into effect.
This patch updates the sequence to update reset cause at first
as part of reset flow.

Change-Id: I2f89ed6316099f310ac86f1701c5fcf2962b4403
---

diff --git a/arch/gaudi2/common/reset.c b/arch/gaudi2/common/reset.c
index 3de8820..5892f7c 100644
--- a/arch/gaudi2/common/reset.c
+++ b/arch/gaudi2/common/reset.c
@@ -86,11 +86,11 @@
 
 void hl_dev_reset(void)
 {
+	hl_set_reset_cause(HL_SW_ALL_RST);
+
 	hl_pcie_disable_access();
 
 	gaudi2_hw_fini();
-
-	hl_set_reset_cause(HL_SW_ALL_RST);
 
 	wfe();
 }
